tests/mock_crust/churn.rs

```diff
@@ -190,18 +228,21 @@ impl ExpectedGets {
             })
             .collect();
         let mut section_msgs_received = HashMap::new(); // The count of received section messages.
+        let mut unexpected_receive = 0;
         for node in nodes {
             while let Ok(event) = node.try_next_ev() {
                 if let Event::Request { request: Request::Get(data_id, msg_id), src, dst } = event {
                     let key = (data_id, msg_id, src, dst);
                     if dst.is_multiple() {
-                        assert!(self.sections
-                                    .get(&key.3)
-                                    .map_or(false, |entry| entry.contains(&node.name())),
-                                "Unexpected request for node {:?}: {:?} / {:?}",
-                                node.name(),
-                                key,
-                                self.sections);
+                        if !self.sections
+                            .get(&key.3)
+                            .map_or(false, |entry| entry.contains(&node.name())) {
+                            trace!("Unexpected request for node {:?}: {:?} / {:?}",
+                                   node.name(),
+                                   key,
+                                   self.sections);
+                            unexpected_receive += 1;
```

Comment:
For `Section` and `PrefixSection` destinations, this should still be impossible, shouldn't it? In that case, let's assert that `dst` is in fact a _group_ authority here.